libs/domain/product/src/mixins/product.mixin.ts

```diff
@@ -34,15 +30,23 @@ export const ProductMixin = <
   class ProductMixinClass extends superClass {
     @signalProperty({ reflect: true }) sku?: string;
 
-    protected productController = new ProductController(this);
+    protected productService = resolve(ProductService, null);
+    protected contextController = new ContextController(this);
+
+    protected $context = signal(
```

Comment:
`$context` could be confusing here, as it's too generic term... what about `$sku`? (I'm aware we already have sku property)...